api/classes/publisher.js

```diff
@@ -201,6 +201,18 @@ class BasePublisher {
     });
   }
 
+  updateProjectReadOnlyProperty(readonly) {
+    if (typeof readonly !== `boolean`) {
+      return Promise.resolve();
+    }
+
+    return projectsQueryBuilder
+    .updateOne(this.project.id, { readonly })
+    .then(() => {
+      this.project.readonly = readonly;
```

Comment:
Want to try to avoid unnecessary calls to the db, also since `this.project` is not a model but the json representation of a model.